File: src/main/java/me/xdrop/nightowl/settings/form/NightOwlSettingsForm.java
package me.xdrop.nightowl.settings.form;

import me.xdrop.nightowl.NightOwlAppearance;
import me.xdrop.nightowl.settings.NightOwlSettings;

import javax.swing.*;

public class NightOwlSettingsForm {
    private JPanel rootPanel;
    private JCheckBox isAppearanceSettingsEnabled;
    private JButton setNow;
    private JLabel toggleTheThemeOnLabel;
    private NightOwlSettings settings;

    public NightOwlSettingsForm() {
        settings = NightOwlSettings.getInstance();
        setNow.addActionListener(e -> {
           NightOwlAppearance.applyIdeSettings();
        });
        reset();
    }

    public boolean isModified() {
        return isAppearanceSettingsEnabled() != settings.isAppearanceSettingsEnabled();
    }

    public void reset() {
        isAppearanceSettingsEnabled.setSelected(settings.isAppearanceSettingsEnabled());
    }

    public JPanel getRootPanel() {
        return rootPanel;
    }

    public boolean isAppearanceSettingsEnabled() {
        return isAppearanceSettingsEnabled.isSelected();
    }

}

File: src/main/kotlin/me/xdrop/nightowl/settings/form/NightOwlSettingsForm.kt
package me.xdrop.nightowl.settings.form

import me.xdrop.nightowl.Appearance
import me.xdrop.nightowl.settings.NightOwlSettings
import javax.swing.JButton
import javax.swing.JCheckBox
import javax.swing.JComponent
import javax.swing.JPanel

class NightOwlSettingsForm {
    private var rootPanel: JPanel? = null
    private var isAppearanceSettingsEnabledCheckBox: JCheckBox? = null
    private var isOverrideRainbowBracketsCheckBox: JCheckBox? = null
    private var setNow: JButton? = null

    private val settings: NightOwlSettings = NightOwlSettings.instance

    init {
        setNow?.addActionListener { Appearance.applyIdeSettings() }
        reset()
    }

    fun component(): JComponent? = rootPanel

    fun reset() {
        isAppearanceSettingsEnabledCheckBox?.isSelected = settings.isAppearanceSettingsEnabled
        isOverrideRainbowBracketsCheckBox?.isSelected = settings.isOverrideRainbowBrackets
    }

    val isAppearanceSettingsEnabled
            get() = isAppearanceSettingsEnabledCheckBox?.isSelected

    val isOverrideRainbowBrackets
        get() = isOverrideRainbowBracketsCheckBox?.isSelected

    val isModified: Boolean
        get() = isAppearanceSettingsEnabled != settings.isAppearanceSettingsEnabled
                || isOverrideRainbowBrackets != settings.isOverrideRainbowBrackets
}
